packages/wonder-blocks-floating/src/components/floating.tsx

```diff
@@ -1,5 +1,208 @@
+    /**
+     * The flip strategy to use.
+     * @default true
+     */
+    flip?: boolean;
+
+    /**
+     * Whether to hide the floating element when the reference element is hidden.
+     * @default true
+     */
+    hide?: boolean;
+
+    /**
+     * The offset of the floating element from the reference element.
+     * @default 20
+     */
+    offset?: number;
+
+    /**
+     * The shift strategy to use.
+     * @default true
+     */
+    shift?: boolean;
+
+    /**
+     * Whether to show the arrow on the floating element.
+     * @default true
+     */
+    showArrow?: boolean;
+
+    // TODO(WB-2111.3): Add props for dismissEnabled, portal and useFocusManager
+};
+
+/**
+ * A component that uses the Floating UI library to position a floating element
+ * relative to a reference element. The floating element appears on hover or
+ * focus.
+ */
+export default function Floating({
+    content,
+    children,
+    boundaryPadding = 8,
+    placement = "top",
+    open = false,
+    onOpenChange,
+    strategy = "absolute",
+    testId,
+    // middleware specific
+    hide: hideProp = true,
+    offset: offsetProp = 20,
+    flip: flipProp = true,
+    shift: shiftProp = true,
+    showArrow = true,
+}: FloatingProps) {
+    const arrowRef = React.useRef(null);
+    const prevOpenRef = React.useRef(open ?? false);
+
+    // Calculate the floating styles and context
+    const {refs, floatingStyles, context, middlewareData} = useFloating({
+        open,
+        onOpenChange,
+        placement,
+        strategy,
+        // Ensure the floating element stays in sync with the reference element
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            // Add offset from the reference element
+            offset({mainAxis: offsetProp}),
+            // Flip to the opposite side if there's not enough space
+            flipProp ? flip() : undefined,
+            // Shift along the axis to keep it in view
+            shiftProp
+                ? shift({
+                      padding: boundaryPadding,
+                      crossAxis: true,
+                  })
+                : undefined,
+            showArrow ? arrow({element: arrowRef}) : undefined,
+            hideProp ? hide() : undefined,
+        ],
+    });
+
+    // call onOpenChange when the floating element is opened or closed
+    React.useEffect(() => {
+        // only trigger when the open value changes and is controlled mode
+        if (prevOpenRef.current !== open) {
+            onOpenChange?.(open);
+            // update the previous open value
+            prevOpenRef.current = open;
+        }
+    }, [onOpenChange, open]);
+
+    // Clone the child element and add the ref
+    const trigger = React.useMemo(() => {
+        return React.cloneElement(children, {
+            ref: refs.setReference,
+        });
+    }, [children, refs.setReference]);
+
+    return (
+        <>
+            {trigger}
```

Comment:
Does the trigger need any a11y attributes? Maybe `aria-haspopup` or `aria-controls` or `aria-expanded`? 

I see Tooltips currently include `aria-describedby` on the trigger, and Popovers include `aria-expanded` and `aria-controls`, so it might be up to the consumer!

Maybe we can provide some docs or guidelines around providing a11y attributes!

Reply:
That should be up to the consumer. I'll add some docs around that as you suggest!.

Reply:
Added the Accessibility page to explain this for now. Let me know if you see something missing.